### PR TITLE
US-15542: Handle navigation on interruption page and declaration page

### PR DESCRIPTION
**Pull request by gautamkumar-hmrc**
Added button on interruption and declaration screen and handled its navigation

TODO - Refactoring the code with adding dependency to useEffect on check answer page and its unit test which we will do as part of enhancement

### CODE REVIEW

File: src/components/custom-sdk/template/HMRC_ODX_GDSCheckAnswersPage/index.tsx [hunk 1]
```diff
@@ -82,20 +82,29 @@ export default function HmrcOdxGdsCheckAnswersPage(props: HmrcOdxGdsCheckAnswers
       });
   }
 
+  useEffect(() => {
+    if (sessionStorage.getItem('isCommingFromEduStartPages') === 'true') {
```
Comment (lcart97) [inline src/components/custom-sdk/template/HMRC_ODX_GDSCheckAnswersPage/index.tsx:86]:
```suggestion
    if (sessionStorage.getItem('isComingFromEduStartPages') === 'true') {
```
Typo

File: src/components/custom-sdk/template/HMRC_ODX_GDSCheckAnswersPage/index.tsx [hunk 2]
```diff
@@ -82,20 +82,29 @@ export default function HmrcOdxGdsCheckAnswersPage(props: HmrcOdxGdsCheckAnswers
...
+      sessionStorage.setItem('isEditMode', 'true');
+      sessionStorage.setItem('isComingFromPortal', 'false');
+      sessionStorage.setItem('isCommingFromEduStartPages', 'false');
```
Comment (lcart97) [inline src/components/custom-sdk/template/HMRC_ODX_GDSCheckAnswersPage/index.tsx:89]:
```suggestion
      sessionStorage.setItem('isComingFromEduStartPages', 'false');
```

File: src/components/override-sdk/infra/ActionButtons/ActionButtons.tsx [hunk 3]
```diff
@@ -31,6 +40,39 @@ export default function ActionButtons(props) {
     thePConn.getActionsApi().navigateToStep(taskListStepId, containerID);
   }
 
+  function navigateToCYA(event) {
+    event.preventDefault();
+    interface ResponseType {
+      CurrentStepId: string;
+    }
```
Comment (lcart97) [inline src/components/override-sdk/infra/ActionButtons/ActionButtons.tsx:47]: Could we define this interface at the top of the file instead?

File: src/components/override-sdk/infra/ActionButtons/ActionButtons.tsx [hunk 4]
```diff
@@ -31,6 +40,39 @@ export default function ActionButtons(props) {
...
+      .getPageDataAsync(
+        'D_GetStepIdByApplicationAndAction',
+        'root',
+        {
+          FlowActionName: 'CheckYourAnswers',
+          CaseID: thePConn.getCaseSummary().content.pyID,
+          ...(isEduStartJourney() && { ApplicationName: 'EDStart' })
+        },
+        options
+      ) // @ts-ignore
```
Comment (lcart97) [inline src/components/override-sdk/infra/ActionButtons/ActionButtons.tsx:63]: Are these  @ts-ignore needed if you have defined a interface already?

File: src/components/custom-sdk/template/HMRC_ODX_GDSCheckAnswersPage/index.tsx [hunk 5]
```diff
@@ -82,20 +82,29 @@ export default function HmrcOdxGdsCheckAnswersPage(props: HmrcOdxGdsCheckAnswers
       });
   }
 
+  useEffect(() => {
+    if (sessionStorage.getItem('isComingFromEduStartPages') === 'true') {
+      sessionStorage.setItem('isEditMode', 'true');
+      sessionStorage.setItem('isComingFromPortal', 'false');
+      sessionStorage.setItem('isComingFromEduStartPages', 'false');
+    }
+  });
```
Comment (lcart97) [inline src/components/custom-sdk/template/HMRC_ODX_GDSCheckAnswersPage/index.tsx:91]:
```suggestion
  }, []);
```
Don't forget to add the dependency array

**Review by monalisasarkar123 — APPROVED** [review]

**Review by akshaykalaskar1 — APPROVED** [review]

**Review by Carl-M-H — APPROVED** [review]

**Review by lcart97 — APPROVED** [review]

**Pull request merged**